### PR TITLE
Check that compatible can connect to mainnet

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR builds upon #8501, but this PR is against `compatible` directly to test that CI works as intended.

This uses the binaries with different signatures generated in #8501 to create different deb packages for testnet and mainnet signatures. All binaries used and packaged by CI use testnet signatures; the new `-mainnet` packages are the mainnet versions, which the `connect-to-mainnet` CI job uses explicitly.

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: buildkite/scripts/build-artifact.sh [hunk 1]
```diff
@@ -35,7 +35,14 @@ echo "--- Upload genesis data"
...
 echo "Building from Commit SHA: $CODA_COMMIT_SHA1"
-dune build --profile=${DUNE_PROFILE} src/app/logproc/logproc.exe src/app/cli/src/mina.exe src/app/rosetta/rosetta.exe 2>&1 | tee /tmp/buildocaml3.log
+dune build --profile=${DUNE_PROFILE} \
+  src/app/logproc/logproc.exe \
+  src/app/cli/src/mina.exe \
+  src/app/cli/src/mina_testnet_signatures.exe \
+  src/app/cli/src/mina_mainnet_signatures.exe \
+  src/app/rosetta/rosetta.exe \
+  src/app/rosetta/rosetta_mainnet_signatures.exe \
+  src/app/rosetta/rosetta_testnet_signatures.exe 2>&1 | tee /tmp/buildocaml3.log
```
Comment (bkase) [inline buildkite/scripts/build-artifact.sh:45]: 👍

File: buildkite/scripts/export-git-env-vars.sh [hunk 2]
```diff
@@ -14,7 +14,13 @@ set +u
 
 # export PVKEYHASH=$(/workdir/_build/default/src/app/cli/src/mina.exe internal snark-hashes | sort | md5sum | cut -c1-8)
 
-export PROJECT="mina-$(echo "$DUNE_PROFILE" | tr _ -)"
+
+# Everything that uses this is doing a testnet_postake_medium_curves build, and
+# we generate packages with different signatures there. Instead of trying to
+# infer a name and ending up with a silly mina-mainnet-testnet package, lets
+# just call them mina-dev.
```
Comment (bkase) [inline buildkite/scripts/export-git-env-vars.sh:21]: 👍

**Review by bkase — APPROVED** [review]
Great! This is awesome that we're generating both types of executables now -- as discussed, worth aligning with @lk86 re any scripts that depend on the mina.exe being mainnet-built

Also, can you change `testnet_postake_medium_curves.mlh` features -> to stop it from pointing to mainnet.mlh (I think this was one of the crazy last minute things near launch day that accidentally landed)

**Comment by mrmr1993** [comment]
Merging compatible seems to make the test fail; currently using CI to bisect on compatible to see what's breaking it so that we can revert.

**Comment by mrmr1993** [comment]
This PR now includes #8584, which in turn depends on #8497. This pull request includes those changes directly in order to test CI against compatible, but those PRs should be merged first.

**Comment by aneesharaines** [comment]
!approved-for-mainnet

**Pull request merged**